src/drivers/adc/adc.c

```diff
@@ -0,0 +1,88 @@
+#include "FreeRTOS.h"
+#include "drivers/adc/adc.h"
+#include "stm32h7xx_hal.h"
+#include <semphr.h>
+
+extern ADC_HandleTypeDef hadc1;
+
+SemaphoreHandle_t adc_conversion_semaphore;
+SemaphoreHandle_t adc_semaphore;
```

Comment:
```suggestion
SemaphoreHandle_t adc_conversion_semaphore = NULL;
SemaphoreHandle_t adc_semaphore = NULL;
```
as per BARR-C:2018 7.2d